torch/dispatch/dispatcher.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch._C as _C
+from torch._C import DispatchKey, DispatchKeySet, ExcludeDispatchKeyGuard  # type: ignore[attr-defined]
+from torch.utils._pytree import tree_flatten
+from torch.overrides import handle_torch_function, has_torch_function
+
+"""
+This is a dispatcher (in Python)
+- You can define new operations (in Python) without schemas
+- It interfaces with the PyTorch dispatcher
+"""
+
+class PyDispatcher:
+    def call(self, operator, args, kwargs):
+        dispatch_key_set = compute_keyset(args, kwargs)
+        kernel = operator.lookup(dispatch_key_set)
+        return call_kernel(kernel, args, kwargs)
+
+    def redispatch(self, operator, dispatch_key_set, args, kwargs):
+        kernel = operator.lookup(dispatch_key_set)
+        return call_kernel(kernel, args, kwargs)
+
+
+dispatcher_singleton = PyDispatcher()
+
+
+def compute_keyset(args, kwargs):
+    tensors = get_tensors(args, kwargs)
+    return key_extractor(tensors)
+
+
+def call_kernel(kernel, args, kwargs):
+    return kernel(*args, **kwargs)
```

Comment:
I'd probably have just inlined this function

Reply:
Done - I had this just for backtrace purposes.